skimage/morphology/_util.py

```diff
@@ -258,13 +263,13 @@ def _resolve_neighborhood(footprint, connectivity, ndim):
                 "match"
             )
         # Must only specify direct neighbors
-        if any(s != 3 for s in footprint.shape):
+        if enforce_adjacency and any(s != 3 for s in footprint.shape):
             raise ValueError("dimension size in footprint is not 3")
```

Comment:
I think we should probably still enforce an odd size along each dimension? Maybe also require that the center element of the footprint is True?

```suggestion
        if not enforce_adjacency and any((s % 2 != 1) for s in footprint.shape):
            raise ValueError("footprint size must be odd along all dimensions")
        elif any(s != 3 for s in footprint.shape):
            raise ValueError("dimension size in footprint is not 3")
```

Reply:
(It is not clear how to define the "center pixel" location if the footprint isn't odd sized)

Reply:
I think line 268 should be `elif enforce_adjacency and any(s != 3 for s in footprint.shape):`. Without the extra `enforce_adjacency` check, in the case where `footprint=np.ones((5, 7))` and `enforce_adjacency=False`, line 266 would evaluate to `False`, and line 268 would evaluate to `True`, resulting in an error being thrown when it shouldn't. I've added a commit that swaps the ordering so that we don't need to check enforce_adjacency twice in the conditions.